### PR TITLE
fix: #RDV-23 fix creation of grid with addind security

### PR DESCRIPTION
**Pull request by niko-amosse**
## Describe your changes

fix creation of grid with addind security

## Issue ticket number and link

[RDV-23](https://jira.support-ent.fr/browse/RDV-23)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)
- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by qperie — APPROVED** [review]
Back ok

File: frontend/src/providers/GridModalProvider/index.tsx [hunk 1]
```diff
@@ -52,22 +53,47 @@ export const GridModalProvider: FC<GridModalProviderProps> = ({ children }) => {
...
+      setInputs,
+      setErrorInputs,
+      structures,
+      existingGridsNames || [],
+    );
 
   const blurGridModalInputs: useBlurGridInputsReturnType = useBlurGridInputs(
     inputs,
     setErrorInputs,
+    existingGridsNames || [],
```
Comment (ThomasSevagen) [inline frontend/src/providers/GridModalProvider/index.tsx:78]: `existingGridsNames ?? []`
va voir la diff

File: frontend/src/providers/GridModalProvider/index.tsx [hunk 2]
```diff
@@ -81,22 +107,17 @@ export const GridModalProvider: FC<GridModalProviderProps> = ({ children }) => {
       setInputs,
       errorInputs,
       setErrorInputs,
+      existingGridsNames: existingGridsNames || [],
       structureOptions: structures,
-      publicOptions: mockPublicList,
+      publicOptions: groups || [],
```
Comment (ThomasSevagen) [inline frontend/src/providers/GridModalProvider/index.tsx:112]: same

File: frontend/src/providers/GridModalProvider/utils.ts [hunk 3]
```diff
@@ -97,7 +90,9 @@ export const gridInputsToGridPayload = (
     structureId: inputs.structure.id,
     duration: formatSlotDurationToString(inputs.slotDuration),
     periodicity: inputs.periodicity,
-    targetPublicListId: inputs.public.map((p) => p.id),
+    targetPublicListId: inputs.public.length
+      ? inputs.public.map((p) => p.id)
```
Comment (ThomasSevagen) [inline frontend/src/providers/GridModalProvider/utils.ts:94]: rename la paramètre du .map, les ia sortent des variables à une lettre mais c'est une mauvaise pratique.
item par défaut sinon quelque chose de plus explicite

File: frontend/src/providers/GridModalProvider/index.tsx [hunk 4]
```diff
@@ -81,22 +107,17 @@ export const GridModalProvider: FC<GridModalProviderProps> = ({ children }) => {
...
     }),
-    [
-      inputs,
-      setInputs,
-      errorInputs,
-      setErrorInputs,
-      structures,
-      mockPublicList,
-    ],
+    [inputs, setInputs, errorInputs, setErrorInputs, structures],
```
Comment (ThomasSevagen) [inline frontend/src/providers/GridModalProvider/index.tsx:120]: dans les deps du memo du provider, on mets les states pas les setStates, et toute autre variable qui nécessite une update

File: frontend/src/providers/GridModalProvider/index.tsx [hunk 5]
```diff
@@ -52,22 +53,47 @@ export const GridModalProvider: FC<GridModalProviderProps> = ({ children }) => {
...
     blurGridModalInputs.handleVisioLinkBlur();
   };
 
+  const resetInputs = () => {
+    setInputs(initialGridModalInputs(structures));
+    setErrorInputs(initialErrorInputs);
+  };
+
+  useEffect(() => {
+    if (inputs.structure.id) {
```
Comment (ThomasSevagen) [inline frontend/src/providers/GridModalProvider/index.tsx:92]: ` if (inputs.structure.id) refetchGroups();`
pas besoin d'accolade

File: frontend/src/providers/GridModalProvider/index.tsx [hunk 6]
```diff
@@ -52,22 +53,47 @@ export const GridModalProvider: FC<GridModalProviderProps> = ({ children }) => {
...
+      existingGridsNames || [],
+    );
 
   const blurGridModalInputs: useBlurGridInputsReturnType = useBlurGridInputs(
     inputs,
     setErrorInputs,
+    existingGridsNames || [],
   );
 
   const updateFirstPageErrors = () => {
```
Comment (ThomasSevagen) [inline frontend/src/providers/GridModalProvider/index.tsx:81]: on destructure plus haut : 
`const {handleNameBlur, handleVisioLinkBlur} = blurGridModalInputs`

File: frontend/src/containers/GridModal/index.tsx [hunk 7]
```diff
@@ -86,8 +88,10 @@ export const GridModal: FC<GridModalProps> = ({ gridModalType }) => {
     )
       return;
 
-    const payload: GridPayload = gridInputsToGridPayload(inputs);
+    const payload: GridPayload = gridInputsToGridPayload(inputs, publicOptions);
```
Comment (ThomasSevagen) [inline frontend/src/containers/GridModal/index.tsx:91]: la gestion d'erreurs???
si ta requete passe pas il se passe quoi??

**Review by ThomasSevagen — APPROVED** [review]

**Pull request merged**